lib/new_relic/agent.rb

```diff
@@ -109,6 +109,7 @@ class SerializationError < StandardError; end
     @logger = nil
     @tracer_lock = Mutex.new
     @tracer_queue = []
+    @metrics_already_recorded = []
```

Comment:
We only add metrics to this array when `#record_metric_once` is called, correct? So the number of items in the object would max out with the number of times we call this method?

Reply:
It would actually max out with the number of times we call this method with a unique metric name. If we call this method 100 times all with the same metric, it would still only be a size of 1.  If we called it with a unique metric name each time though, that would increase it, yes.